planner/nodes/planner.py

```diff
@@ -137,17 +140,12 @@ def planner(datapackage_input, processing, outputs, allowed_types=None):
                         'stream': True
                     })
                 )
+                pipeline_steps.extend(required_artifact.pipeline_steps)
             else:
-                pipeline_steps.append(
-                    ('add_resource', ri)
-                )
-                needs_streaming = True
-
-        if needs_streaming:
-            pipeline_steps.extend([
-                ('assembler.sample',),
-                ('stream_remote_resources',)
-            ])
+                pipeline_steps.extend([
+                    ('add_resource', ri),
+                    ('stream_remote_resources',)
```

Comment:
This used to be outside of the for loop, and now it's inside... why?

Reply:
@akariv to run pipeline_steps for each `required_streamed_artifacts`

Eg we want to add built-in `sort` pipeline and only derived/csv is needed: 
* whenever source/tabular comes as `required_streamed_artifacts` for derived/csv it has it's own pipeline_steps (in this case `sort` )
* pipeline_steps for source/tabular artefact are never executed unless so I decided to execute right after adding resource
* `sort` will not be executed unless data is streamed
* if `required_streamed_artifacts` are more than one we need to add `sort` for each of them (reason inside loop)
* So since pipeline_steps already included `stream_remote_resources` I decided to remove from outside the loop as it seemed extra

Reply:
In case of multiple resources, there's no need to repeat `stream_remote_resources` - it's redundant (as `stream_remote_resources` will stream all remote resources...).
Might be better to have one `stream_remote_resources` and put all processing steps afterwards.